### PR TITLE
Update logging guidance

### PR DESCRIPTION
**Pull request by soupdragon99**
Update guidance to reflect GOV.UK style.

### CODE REVIEW

File: source/standards/logging.html.md.erb [hunk 1]
```diff
@@ -1,108 +1,47 @@
...
+* non-production environments - no more than 7 days
+* production environments - no more than 30 days
 
-Each team should not operate its own ELK stack for logging.
-In the past we've found that ELK is relatively easy to set up,
-but teams tend to not prioritise keeping it updated.
+### Long term archiving
 
-However, ELK isn't appropriate for long term archival of logs. There
-are other services which are less expensive and more reliable.
```
Comment (soupdragon99) [inline source/standards/logging.html.md.erb]: we could do with an example or two of services which are less expensive and more reliable.

Comment (soupdragon99) [reply source/standards/logging.html.md.erb]: @stephenharker @tomnatt can you suggest something here?

File: source/standards/logging.html.md.erb [hunk 2]
```diff
@@ -1,108 +1,47 @@
...
-In general, tools for querying logs are not a good fit for long-term
-archival, and vice versa.
+You shouldn’t operate your own self hosted ELK stacks for logging. In the past we’ve found ELK relatively easy to set up, but long term it’s easier to use a cloud-hosted ELK solution.
 
-### Shared Elasticsearch, Logstash and Kibana (ELK) stack
+You should be aware that ELK isn’t appropriate for long term archival of logs. There are other services which are less expensive and more reliable.
 
-GDS has a shared account with [logit](https://logit.io/), which offers
-on-demand ELK stacks as a service.
+You can read the [Reliability Engineering team documentation](https://reliability-engineering.cloudapps.digital/#logging) for more information about using Logit.
```
Comment (tomnatt) [inline source/standards/logging.html.md.erb]: Can we move this to be with the bit about Logit 3 paragraphs earlier please?

File: source/standards/logging.html.md.erb [hunk 3]
```diff
@@ -1,108 +1,47 @@
...
 
 <%= partial :expires %>
 
-## Short-term querying of logs
+For incident management and debugging you should have a system in place so you can store and query your infrastructure and application logs.
 
-For the purposes of incident management and debugging, it's useful to
-have a system for storing and querying the most recent logs produced
-by your application and infrastructure.
+GDS has a shared account with [Logit](https://logit.io/), offering on-demand shared [ELK (Elasticsearch, Logstash and Kibana)](https://www.elastic.co/elk-stack) stacks as a service. ELK offers suitable short-term storage for your logs so they are both accessible and queryable.
```
Comment (tomnatt) [inline source/standards/logging.html.md.erb]: This should be more decisive. Teams should be moving to the central Logit install by default.

Also, I think that second ELK (ELK offers...) should be Logit for clarity.

File: source/standards/logging.html.md.erb [hunk 4]
```diff
@@ -1,108 +1,47 @@
...
+* production environments - no more than 30 days
 
-Each team should not operate its own ELK stack for logging.
-In the past we've found that ELK is relatively easy to set up,
-but teams tend to not prioritise keeping it updated.
+### Long term archiving
 
-However, ELK isn't appropriate for long term archival of logs. There
-are other services which are less expensive and more reliable.
+When storing logs you should focus on long-term durability. Some GDS teams archive logs to [Amazon S3](https://aws.amazon.com/s3/) and at the same time send them to their ELK or [Sumo Logic](https://www.sumologic.com/) store.
```
Comment (tomnatt) [inline source/standards/logging.html.md.erb]: Didn't we just advise teams to not have their own ELK instances?

File: source/standards/logging.html.md.erb [hunk 5]
```diff
@@ -1,108 +1,47 @@
...
-It's a good idea to be able to take logs from the long-term archive
-and load them into a queryable tool (for example, a one-time-use ELK
-stack).
+* how your logs will get to short and long-term stores?
+* what happens if one of these stores is unavailable?
+* whether the store will be overloaded when it comes back online?
 
-If you are using Amazon Web Services, CloudWatch can run rudimentary
-queries on older logs which may be good enough.
+You can use [Elastic Beats](https://www.elastic.co/products/beats) family of tools which detect periods of unavailability. They remember how much data they send and continue where they left off. By using a protocol which is backpressure-aware they also won’t overload [Logstash](https://www.elastic.co/products/logstash) when it comes back online.
```
Comment (tomnatt) [inline source/standards/logging.html.md.erb]: You can use the Elastic Beats....

File: source/index.html.md.erb [hunk 6]
```diff
@@ -53,7 +53,7 @@ These standards will enable us to maintain a consistent operating environment ac
 - [operating systems](standards/operating-systems.html)
 - [responding to problems](standards/alerting.html)
 - [how to send email notifications](standards/sending-email.html)
-- [storing and querying logs](standards/logging.html)
+- [store and query logs](standards/logging.html)
```
[the remark this answers is not in the document]
Comment (soupdragon99) [reply source/index.html.md.erb]: There was some talk of having logging guidance etc. in one section but I think that's a way off. I'd rather leave as is for now. I know the guidance has a title of logging, but this piece of content only covers storage and querying and not 'logging' in general or Logit etc.

File: source/standards/logging.html.md.erb (hunk 2 above)
Comment (soupdragon99) [reply source/standards/logging.html.md.erb]: Agreed.

File: source/standards/logging.html.md.erb (hunk 3 above)
Comment (soupdragon99) [reply source/standards/logging.html.md.erb]: Agreed

File: source/standards/logging.html.md.erb (hunk 4 above)
Comment (soupdragon99) [reply source/standards/logging.html.md.erb]: I've changed this now, and also moved a statement we make about ELK not being appropriate for long term storage to this section.

File: source/standards/logging.html.md.erb (hunk 5 above)
Comment (soupdragon99) [reply source/standards/logging.html.md.erb]: Agreed

**Comment by soupdragon99** [comment]
Content is currently being 2i reviewed.

**Pull request merged**